src/main/java/net/dv8tion/jda/core/requests/WebSocketClient.java

```diff
@@ -415,9 +417,16 @@ public void onConnected(WebSocket websocket, Map<String, List<String>> headers)
         messagesSent = 0;
         ratelimitResetTime = System.currentTimeMillis() + 60000;
         if (sessionId == null)
-            sendIdentify();
+        {
+            if (!firstInit && reconnectQueue != null)
+                reconnectQueue.appendSession(this);
```

Comment:
Won't this run into an issue where if a lot of shards attempt to reconnect, and they are all placed in this queue, then shards at the end of the queue have to wait for a significant amount of time, thus their period for identify/resume will expire.

What happens if the MainWS DCs and then the reconnectQueue attempts to sendIdentify?

Reply:
I didn't think much about the implementation because I wanted a fix fast for now, I'm looking further into it.

Reply:
The sessions are only added to the reconnect queue if they cannot resume. Active sessions will not be queued but full new setup will require them to wait their turn.
The system is designed to first come first served using a blocking queue that is handled when it is first populated.
Reconnect will happen on the queue worker causing it to lose the identification string which is appended to the client's threads, instead it logs the shard information when it reconnects the session.
### Exmaple
```
[06:20:39] [Warning] [JDASocket]: Got disconnected from WebSocket (Internet?!)... Appending session to reconnect queue
[06:20:39] [Warning] [JDASocket]: Got disconnected from WebSocket (Internet?!)... Appending session to reconnect queue
[06:20:39] [Warning] [JDASocket]: Queue is attempting to reconnect a shard... Shard: [0 / 2]
[06:20:39] [Warning] [JDASocket]: Attempting to reconnect in 2s
[06:20:41] [Warning] [JDASocket]: Attempting to reconnect!
[06:20:41] [Info] [JDASocket]: Connected to WebSocket
[06:20:42] [Info] [JDA]: Finished (Re)Loading!
[06:20:46] [Warning] [JDASocket]: Queue is attempting to reconnect a shard... Shard: [1 / 2]
[06:20:46] [Warning] [JDASocket]: Attempting to reconnect in 2s
[06:20:48] [Warning] [JDASocket]: Attempting to reconnect!
[06:20:48] [Info] [JDASocket]: Connected to WebSocket
[06:20:48] [Info] [JDA]: Finished (Re)Loading!
```